python/paddle/fluid/imperative/layers.py

```diff
@@ -189,11 +196,19 @@ def __getattr__(self, name):
             return self._sub_layers[name]
 
     def __setattr__(self, name, value):
+        def remove_from_layers(dict):
+            if dict is None:
+                return
+            else:
+                if name in dict:
+                    del dict[name]
+
         if isinstance(value, framework.Parameter):
             params = self.__dict__.get('_parameters', None)
             if params is None:
                 raise ValueError(
                     "super(YourLayer, self).__init__() should be called first")
+            remove_from_layers(self.__dict__.get('_parameters', None))
```

Comment:
why? the parameter might be referenced somewhere else?

Reply:
in that case, parameter should be in the parameter list of the layer which truly hold it, here I just want to make sure each layer hold their own parameters after weight sharing